projects/plugins/jetpack/_inc/client/search/dashboard/module-control.jsx

```diff
@@ -0,0 +1,153 @@
+
+/**
+ * Search settings component to be used within the Performance section.
+ *
+ * @param  {object} props - Component properties.
+ * @returns {React.Component}	Search settings component.
+ */
+function Search( props ) {
+	const { failedToEnableSearch, hasActiveSearchPurchase, updateOptions } = props;
+	const isModuleEnabled = props.getOptionValue( 'search' );
+	const isInstantSearchEnabled = props.getOptionValue( 'instant_search_enabled', 'search' );
+
+	const toggleSearchModule = useCallback( () => {
+		const newOption = { search: ! isModuleEnabled };
+		if ( isInstantSearchEnabled !== ( hasActiveSearchPurchase && ! isModuleEnabled ) ) {
+			newOption.instant_search_enabled = hasActiveSearchPurchase && ! isModuleEnabled;
+		}
+		updateOptions( newOption );
+	}, [ hasActiveSearchPurchase, isInstantSearchEnabled, isModuleEnabled, updateOptions ] );
+
+	const toggleInstantSearch = useCallback( () => {
+		const newOption = {
+			instant_search_enabled: hasActiveSearchPurchase && ! isInstantSearchEnabled,
+		};
+		if ( newOption.instant_search_enabled && ! isModuleEnabled ) {
+			newOption.search = true;
+		}
+		updateOptions( newOption );
+	}, [ hasActiveSearchPurchase, isInstantSearchEnabled, isModuleEnabled, updateOptions ] );
+
+	useEffect( () => {
+		if ( failedToEnableSearch && hasActiveSearchPurchase ) {
+			updateOptions( { has_jetpack_search_product: true } );
+			toggleSearchModule();
+		}
+	}, [ failedToEnableSearch, hasActiveSearchPurchase, updateOptions, toggleSearchModule ] );
+
+	const togglingModule = !! props.isSavingAnyOption( 'search' );
+	const togglingInstantSearch = !! props.isSavingAnyOption( 'instant_search_enabled' );
+	const isSavingEitherOption = togglingModule || togglingInstantSearch;
+
+	return (
+		<Fragment>
+			<QuerySite />
+			<SettingsGroup
+				disableInOfflineMode
+				hasChild
+				module={ { module: 'search' } }
+				support={ {
+					text: SEARCH_SUPPORT,
+					link: getRedirectUrl( 'jetpack-support-search' ),
+				} }
+				className={ [ 'jp-form-search-settings-group' ] }
+			>
+				<p>{ props.inOfflineMode && __( 'Unavailable in Offline Mode', 'jetpack' ) }</p>
+
+				{ props.isLoading && __( 'Loading…', 'jetpack' ) }
+
+				{ ! props.isLoading && ( props.isBusinessPlan || props.hasActiveSearchPurchase ) && (
+					<Fragment>
+						<div className="jp-search-search-toggle jp-search-search-toggle--search">
+							<ModuleToggle
+								activated={ isModuleEnabled }
+								compact
+								disabled={ isSavingEitherOption }
+								slug="search"
+								toggleModule={ toggleSearchModule }
+								toggling={ togglingModule }
+							>
+								{ __( 'Enable Jetpack Search', 'jetpack' ) }
+							</ModuleToggle>
+							<div className="jp-search-search-toggle__description">
+								<p className="jp-form-search-setting__explanation">{ SEARCH_DESCRIPTION }</p>
+							</div>
+						</div>
+						<div className="jp-search-search-toggle jp-search-search-toggle--instant-search">
```

Comment:
We've got a mixture of underscore and dash separators in here - `jp-search-search-toggle__description` vs `jp-search-search-toggle--instant-search` - can we use underscores consistently after `jp-search-search-toggle`?

Reply:
`jp-search-search-toggle--instant-search` is meant to be a block modifier while `jp-form-search-setting__explanation` is an element, which is the reason that the naming is different according to [BEM](http://getbem.com/naming/). Does that make sense?

Reply:
Thanks for explaining @kangzj! BEM seems like a good idea but we should clarify what the right CSS standards to follow here are.

I think generally speaking we've used the Calypso CSS guidelines for the Instant Search app itself:

https://github.com/Automattic/wp-calypso/blob/trunk/docs/coding-guidelines/css.md

...and Jetpack just follows the WordPress Core CSS guidelines by the look of the docs:

https://developer.wordpress.org/coding-standards/wordpress-coding-standards/css/

Is that correct @jeherve?